main.nf

```diff
@@ -85,13 +100,15 @@ if (metadata) {
     metadata_to_report
   }.splitCsv(header: true, sep: '\t').into {
     metadata_to_check
-    metadata_to_merge
+    metadata_to_merge_assembly
```

Comment:
Following my comment on option names, I propose these variable names:
`metadata_to_merge_for_assembly`
`metadata_to_merge_for_quantification`
Same goes for all other variables and process names, e.g.:
`groups_to_merge_for_assembly`
`groups_to_log_for_quantification`
etc.

Open to other suggestions, just want to make it homogeneous and understandable at first sight.

PS: When we move to DSL2 we'll just use more generic names for reusable variables.